tests/happy_path/test_node-operators-registry-happy-path.py

```diff
@@ -0,0 +1,568 @@
+INITIAL_TOKEN_HOLDER = "0x000000000000000000000000000000000000dead"
+TOTAL_BASIS_POINTS = 10000
+ZERO_HASH = bytes([0] * 32)
+ZERO_BYTES32 = HexBytes(ZERO_HASH)
+
+
+@pytest.fixture()
+def extra_data_service():
+    return ExtraDataService()
+
+
+@pytest.fixture(scope="module")
+def voting_eoa(accounts):
+    return accounts.at(contracts.voting.address, force=True)
+
+
+StakingModuleId = NewType('StakingModuleId', int)
+NodeOperatorId = NewType('NodeOperatorId', int)
+NodeOperatorGlobalIndex = Tuple[StakingModuleId, NodeOperatorId]
+
+
+def node_operator(module_id, node_operator_id) -> NodeOperatorGlobalIndex:
+    return module_id, node_operator_id
+
+
+@pytest.fixture(scope="module")
+def nor(interface):
+    return interface.NodeOperatorsRegistry(contracts.node_operators_registry.address)
+
+
+def calc_no_rewards(
+    nor, no_id, report_shares
+):
+    total_signing_keys = nor.getNodeOperatorSummary(no_id)[
+        'totalDepositedValidators'] - nor.getNodeOperatorSummary(no_id)['totalExitedValidators']
+    total_deposite_keys = nor.getStakingModuleSummary()[
+        'totalDepositedValidators'] - nor.getStakingModuleSummary()['totalExitedValidators']
+
+    return report_shares // 2 * total_signing_keys // total_deposite_keys
+
+
+def increase_limit(nor, first, second, base, keys_count, voting_eoa):
+    current_first_keys = max(nor.getNodeOperator(first, True)['totalVettedValidators'], nor.getNodeOperator(
+        first, True)['totalAddedValidators'])
+    current_second_keys = max(nor.getNodeOperator(second, True)['totalVettedValidators'], nor.getNodeOperator(
+        second, True)['totalAddedValidators'])
+    current_base_keys = max(nor.getNodeOperator(base, True)['totalVettedValidators'], nor.getNodeOperator(
+        base, True)['totalAddedValidators'])
+
+    nor.setNodeOperatorStakingLimit(
+        first, current_first_keys + keys_count, {'from': voting_eoa})
+    nor.setNodeOperatorStakingLimit(
+        second, current_second_keys + keys_count, {'from': voting_eoa})
+    nor.setNodeOperatorStakingLimit(
+        base, current_base_keys + keys_count, {'from': voting_eoa})
+
+
+def deposit_and_check_keys(nor, first_no_id, second_no_id, base_no_id, keys_count):
+
+    deposited_keys_first_before = nor.getNodeOperatorSummary(
+        first_no_id)['totalDepositedValidators']
+    deposited_keys_second_before = nor.getNodeOperatorSummary(
+        second_no_id)['totalDepositedValidators']
+    deposited_keys_base_before = nor.getNodeOperatorSummary(
+        base_no_id)['totalDepositedValidators']
+
+    module_total_deposited_keys_before = nor.getStakingModuleSummary()[
+        'totalDepositedValidators']
+
+    contracts.lido.deposit(keys_count, 1, '0x', {
+        'from': contracts.deposit_security_module.address})
+
+    module_total_deposited_keys_after = nor.getStakingModuleSummary()[
+        'totalDepositedValidators']
+
+    assert module_total_deposited_keys_before < module_total_deposited_keys_after
```

Comment:
can check for a precise value using the emitted `DepositedValidatorsChanged` and `Unbuffered` events as well as using `DEPOSITED_VALIDATORS` (`Lido.getBeaconStat()`)

Reply:
Added check Unbuffered ETH by keys count